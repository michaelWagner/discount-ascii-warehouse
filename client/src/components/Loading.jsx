import React from 'react';

class Loading extends React.Component {
  constructor() {
    super();
  }

  render() {
    return (
      <div className={this.props.hasLoaded}>
        <span className="loader">
          <div className="loader-indicator" >
            <h1>
              <span>Loading</span>
              <span className="loader-ellipsis" >
                <span className="loader-ellipsisDot">.</span>
                <span className="loader-ellipsisDot">.</span>
                <span className="loader-ellipsisDot">.</span>
              </span>
            </h1>
          </div>
        </span>
      </div>
    );
  }
}

export default Loading;
