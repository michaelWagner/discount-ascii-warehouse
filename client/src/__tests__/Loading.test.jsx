import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Loading from '../components/Loading';

describe(Loading, () => {

  const parentComponentName = 'ProductGrid';
  const loadState = 'loading-bar';
  const component = shallow(
    <Loading hasLoaded={loadState}/>
  );

  it('renders and matches our snapshot', () => {
    const component = renderer.create(
      <Loading hasLoaded={loadState}/>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
