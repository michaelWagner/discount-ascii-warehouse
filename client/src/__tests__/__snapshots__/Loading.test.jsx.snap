exports[`class Loading extends /* istanbul ignore next */_react2.default.Component {

  render() {
    return (
      /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */"div", /* istanbul ignore next */{ className: this.props.hasLoaded, __source: { fileName: _jsxFileName, lineNumber: 7 }, __self: this },
        /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */"span", /* istanbul ignore next */{ className: "loader", __source: { fileName: _jsxFileName, lineNumber: 8 }, __self: this },
          /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */"div", /* istanbul ignore next */{ className: "loader-indicator", __source: { fileName: _jsxFileName, lineNumber: 9 }, __self: this },
            /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */"h1", /* istanbul ignore next */{ __source: { fileName: _jsxFileName, lineNumber: 10 }, __self: this },
              /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */"span", /* istanbul ignore next */{ __source: { fileName: _jsxFileName, lineNumber: 11 }, __self: this }, /* istanbul ignore next */"Loading"),
              /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */"span", /* istanbul ignore next */{ className: "loader-ellipsis", __source: { fileName: _jsxFileName, lineNumber: 12 }, __self: this },
                /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */"span", /* istanbul ignore next */{ className: "loader-ellipsisDot", __source: { fileName: _jsxFileName, lineNumber: 13 }, __self: this }, /* istanbul ignore next */"."),
                /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */"span", /* istanbul ignore next */{ className: "loader-ellipsisDot", __source: { fileName: _jsxFileName, lineNumber: 14 }, __self: this }, /* istanbul ignore next */"."),
                /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */"span", /* istanbul ignore next */{ className: "loader-ellipsisDot", __source: { fileName: _jsxFileName, lineNumber: 15 }, __self: this }, /* istanbul ignore next */".")))))));






  }} renders and matches our snapshot 1`] = `
<div
  className="loading-bar">
  <span
    className="loader">
    <div
      className="loader-indicator">
      <h1>
        <span>
          Loading
        </span>
        <span
          className="loader-ellipsis">
          <span
            className="loader-ellipsisDot">
            .
          </span>
          <span
            className="loader-ellipsisDot">
            .
          </span>
          <span
            className="loader-ellipsisDot">
            .
          </span>
        </span>
      </h1>
    </div>
  </span>
</div>
`;
